internal/api/routes_heresphere.go

```diff
@@ -0,0 +1,1402 @@
+
+					typeName := transNames[i]
+					mediaTypes[typeName] = append(mediaTypes[typeName], processedEntry)
+				}
+			}
+		}
+	}
+
+	// Reconstruct tables
+	for codec, sources := range mediaTypes {
+		processedMedia = append(processedMedia, HeresphereVideoMedia{
+			Name:    codec,
+			Sources: sources,
+		})
+	}
+
+	return processedMedia
+}
+
+/*
+ * This endpoint provides the main libraries that are available to browse.
+ */
+func (rs heresphereRoutes) HeresphereIndex(w http.ResponseWriter, r *http.Request) {
+	// Banner
+	banner := HeresphereBanner{
+		Image: fmt.Sprintf("%s%s",
+			GetBaseURL(r),
+			"/apple-touch-icon.png",
+		),
+		Link: fmt.Sprintf("%s%s",
+			GetBaseURL(r),
+			"/",
+		),
+	}
+
+	// Read scenes
+	var scenes []*models.Scene
+	if err := txn.WithReadTxn(r.Context(), rs.repository.TxnManager, func(ctx context.Context) error {
+		var err error
+		scenes, err = rs.repository.Scene.All(ctx)
+		return err
+	}); err != nil {
+		http.Error(w, "Failed to fetch scenes!", http.StatusInternalServerError)
+		return
+	}
+
+	// Create scene list
+	sceneUrls := make([]string, len(scenes))
+	for idx, scene := range scenes {
+		sceneUrls[idx] = fmt.Sprintf("%s/heresphere/%d",
+			GetBaseURL(r),
+			scene.ID,
+		)
+	}
+
+	// All library
+	library := HeresphereIndexEntry{
+		Name: "All",
+		List: sceneUrls,
+	}
+	// Index
+	idx := HeresphereIndex{
+		Access:  HeresphereMember,
+		Banner:  banner,
+		Library: []HeresphereIndexEntry{library},
+	}
+
+	// Create a JSON encoder for the response writer
+	w.Header().Set("Content-Type", "application/json")
+	enc := json.NewEncoder(w)
+	enc.SetEscapeHTML(false)
+	if err := enc.Encode(idx); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+}
+
+/*
+ * This auxiliary function finds vr projection modes from tags and the filename.
+ */
+func FindProjectionTags(scene *models.Scene, processedScene *HeresphereVideoEntry) {
+	// Detect VR modes from tags
+	for _, tag := range processedScene.Tags {
+		tagPre := strings.TrimPrefix(tag.Name, "Tag:")
+
+		// Has degrees tag
+		if strings.HasSuffix(tagPre, "°") {
+			deg := strings.TrimSuffix(tagPre, "°")
+			if s, err := strconv.ParseFloat(deg, 64); err == nil {
+				processedScene.Fov = s
+			}
+		}
+		// Has VR tag
+		vrTag, err := getVrTag()
+		if err == nil && tagPre == vrTag {
+			if processedScene.Projection == HeresphereProjectionPerspective {
+				processedScene.Projection = HeresphereProjectionEquirectangular
+			}
+			if processedScene.Stereo == HeresphereStereoMono {
+				processedScene.Stereo = HeresphereStereoSbs
+			}
+		}
+		// Has Fisheye tag
+		if tagPre == "Fisheye" {
+			processedScene.Projection = HeresphereProjectionFisheye
+			if processedScene.Stereo == HeresphereStereoMono {
+				processedScene.Stereo = HeresphereStereoSbs
+			}
+		}
+	}
+
+	// Detect VR modes from filename
+	file := scene.Files.Primary()
+	if file != nil {
+		path := strings.ToUpper(file.Basename)
```

Comment:
This whole block should go into a separate function.

Reply:
Done!